qt-python: Refactor Python environment handling

benchoq opened the pull request:
Refactor the `PySideEnv` class to have all relevant data and methods for improved maintainability.
Adjust the responsibilities of related classes such as `PySideProject`, `PySideProjectManager` and others accordingly.

<!---
# Qt contribution guidelines

We welcome contributions to Qt!

Read the
[Qt Contribution Guidelines](https://wiki.qt.io/Qt_Contribution_Guidelines) to learn more.
<!---

OrkunTokdemir commented on qt-python/src/project.ts:
```diff
@@ -1,38 +1,47 @@
   }
 
   dispose() {
     logger.info(`Dispose: "${this._folder.uri.fsPath}"`);
```
Can you also dispose `_env` here?
```suggestion
    logger.info(`Dispose: "${this._folder.uri.fsPath}"`);
    this._env.dispose();
```

benchoq replied:
Thanks for noticing that!

benchoq committed:
qt-python: Refactor Python environment handling

Refactor the `PySideEnv` class to have all relevant data and methods
for improved maintainability.
Adjust the responsibilities of related classes such as `PySideProject`,
`PySideProjectManager` and others accordingly.
